web_scraper/web_scraper/spiders/cve_spiders.py

```diff
@@ -0,0 +1,92 @@
+import scrapy
+import os
+
+
+class QuotesSpider(scrapy.Spider):
```

Comment:
It looks like this class is suitable for fetch data from the type of web page with url starts with `https://ubuntu.com/security/CVE-`. If we pass in any other url for example `https://www.youtube.com`, then this class will not work for its purpose. This means that each spider class will be responsible for one type of webpage (which is a good idea i believe). Here are some recommendations because of the 1:1 relationship of the spider class and the type of webpage.

1. Using a class level docstring to explain that this is used to retrieve information from `https://ubuntu.com/security/CVE-` webpages. Also, from the docstring, state the output file format.
2. After getting the `SCRAPE_URL_LIST` from the environment variable, validate the url format.

On a side note: Could you comment on why this class is called QuotesSpider as opposed to something like UbuntuCVEPageSpider?

Reply:
After think about this again, we could also reuse this class to crawl other webpages as well, we are likely need a mapping like below in that case.

```
{
      "https://ubuntu.com/security/CVE-": parse_ubuntu_cve_page,
       ....
}
```